lib/mapping/CCampaignHandler.h

```diff
@@ -221,7 +226,13 @@ class DLL_LINKAGE CCampaignHandler
 	std::vector<size_t> scenariosCountPerCampaign;
 
 	static std::string readLocalizedString(CBinaryReader & reader, std::string filename, std::string modName, std::string encoding, std::string identifier);
+	
+	//parsers for VCMI campaigns (*.vcmp)
+	static CCampaignHeader readHeaderFromJson(JsonNode & reader, std::string filename, std::string modName, std::string encoding);
+	static CCampaignScenario readScenarioFromJson(JsonNode & reader, std::string filename, std::string modName, std::string encoding, int version, int mapVersion );
+	static CScenarioTravel readScenarioTravelFromJson(JsonNode & reader, int version);

```

Comment:
We really need stateful campaign handler/parser instead of this static forestry

(not necessarily in this PR)

Reply:
We may consider adding this PR into 1.2 because it's save (h3c maps follows same code as earlier)
It's could be useful for Antagarich Burning campaign

For now idea is to freeze a vcmi file format. Internally for now it's transformed into "h3c format".
Later we should do vice versa